tests/sentry/api/serializers/rest_framework/test_list.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import
+from sentry.api.serializers.rest_framework import ListField
+
+from rest_framework import serializers
+
+from sentry.testutils import TestCase
+
+
+class ListFieldTest(TestCase):
+    def assert_success(self, serializer, value):
+        assert serializer.is_valid()
+        assert not serializer.errors
+        assert serializer.object == value
+
+    def assert_unsuccessful(self, serializer, errors):
+        assert not serializer.is_valid()
+        assert serializer.errors == errors
+        assert serializer.object is None
+
+    def test_simple(self):
+        class DummySerializer(serializers.Serializer):
+            list_field = ListField(child=serializers.IntegerField())
+        serializer = DummySerializer(data={'list_field': [1, 2, 3]})
+        self.assert_success(serializer, {'list_field': [1, 2, 3]})
+
+    def test_single_element_list(self):
+        class DummySerializer(serializers.Serializer):
+            list_field = ListField(child=serializers.IntegerField())
+        serializer = DummySerializer(data={'list_field': [1]})
+        self.assert_success(serializer, {'list_field': [1]})
+
+    def test_empty_list_child_required(self):
+        class DummySerializer(serializers.Serializer):
+            list_field = ListField(required=False, child=serializers.IntegerField())
+        serializer = DummySerializer(data={'list_field': []})
+        self.assert_success(serializer, {'list_field': []})
```

Comment:
Note that this test succeeds. But when a complex serializer is created in a test below it does not.